src/analysis/ring_seminorm.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2022 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import analysis.seminorm
+
+/-!
+# Seminorms and norms on rings
+
+This file defines seminorms and norms on rings. These definitions are useful when one needs to
+consider multiple (semi)norms on a given ring.
+
+## Main declarations
+
+For a ring `R`:
+* `ring_seminorm`: A seminorm on a ring `R` is a function `f : R → ℝ` that preserves zero, takes
+  nonnegative values, is subadditive and submultiplicative and such that `f (-x) = f x` for all
+  `x ∈ R`.
+* `ring_norm`: A seminorm `f` is a norm if `f x = 0` if and only if `x = 0`.
+
+## References
+
+* [S. Bosch, U. Güntzer, R. Remmert, *Non-Archimedean Analysis*][bosch-guntzer-remmert]
+
+## Tags
+ring_seminorm, ring_norm
+-/
+
+noncomputable theory
+
+set_option old_structure_cmd true
+
+open_locale nnreal
+
+variables {R S : Type*} (x y : R) (r : ℝ)
+
+/-- A seminorm on a ring `R` is a function `f : R → ℝ` that preserves zero, takes nonnegative
+  values, is subadditive and submultiplicative and such that `f (-x) = f x` for all `x ∈ R`. -/
+structure ring_seminorm (R : Type*) [ring R]
```

Comment:
I assume this works fine with
```suggestion
structure ring_seminorm (R : Type*) [non_unital_ring R]
```
which is useful because then you can use this structure with [non_unital_semi_normed_ring](https://leanprover-community.github.io/mathlib_docs/analysis/normed/field/basic.html#non_unital_semi_normed_ring)

Reply:
I've made this change and replaced `ring` by `non_unital_ring` wherever possible in the file.